tests/TopDownProteomics.Tests/ProForma/ProFormaWriterTests.cs

```diff
@@ -55,41 +55,51 @@ public void WriteMultipleDescriptors()
             });
             var result = _writer.WriteString(term);
 
-            Assert.AreEqual("SEQ[info:test|mass:+14.05]UENCE", result);
+            Assert.AreEqual("SEQ[Info:test|+14.05]UENCE", result);
         }
 
         [Test]
         public void WriteAmbiguousPossibleSitesDescriptors()
         {
-            var term = new ProFormaTerm("SEQUENCE", tags: new[]
+            var term = new ProFormaTerm("SEQUENCE", tagGroups: new[]
             {
-                new ProFormaTag(2, new[] 
+                new ProFormaTagGroup("test", ProFormaKey.Mass, "+14.05", new[]
                 {
-                    new ProFormaDescriptor(ProFormaKey.Mass, "+14.05"),
-                    new ProFormaAmbiguityDescriptor("#", "test")
+                    new ProFormaMembershipDescriptor(2),
+                    new ProFormaMembershipDescriptor(5),
                 }),
-                new ProFormaTag(5, new[] { new ProFormaAmbiguityDescriptor("#", "test") }),
             });
             var result = _writer.WriteString(term);
 
-            Assert.AreEqual("SEQ[mass:+14.05|#test]UEN[#test]CE", result);
+            Assert.AreEqual("SEQ[+14.05#test]UEN[#test]CE", result);
+
+            // With weights
+            term = new ProFormaTerm("SEQUENCE", tagGroups: new[]
+            {
+                new ProFormaTagGroup("test", ProFormaKey.Mass, "+14.05", new[]
+                {
+                    new ProFormaMembershipDescriptor(2, 0.9),
+                    new ProFormaMembershipDescriptor(5, 0.1),
+                }),
+            });
+            result = _writer.WriteString(term);
+
+            Assert.AreEqual("SEQ[+14.05#test(0.9)]UEN[#test(0.1)]CE", result);
         }
 
         [Test]
         public void WriteAmbiguousRangeDescriptors()
         {
             var term = new ProFormaTerm("SEQUENCE", tags: new[]
             {
-                new ProFormaTag(2, new[]
+                new ProFormaTag(2, 5, new[]
                 {
                     new ProFormaDescriptor(ProFormaKey.Mass, "+14.05"),
-                    new ProFormaAmbiguityDescriptor("->", "test")
-                }),
-                new ProFormaTag(5, new[] { new ProFormaAmbiguityDescriptor("<-", "test") }),
+                })
             });
             var result = _writer.WriteString(term);
 
```

Comment:
Hah, it's fun seeing the arrows one last time. The parentheses are definitely more interpretable.

Reply:
ya, it had a certain appeal, but parens are better